src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala

```diff
@@ -244,7 +269,38 @@ class ErgoMemPool private[mempool](pool: OrderedTxPool,
   def getReader: ErgoMemPoolReader = this
 }
 
-object ErgoMemPool {
+object ErgoMemPool extends ScorexLogging {
+
+  /**
+   * Hierarchy of sorting strategies for mempool transactions
+   */
+  sealed trait SortingOption
+
+  object SortingOption {
+    /**
+      * Sort transactions by fee paid for transaction size, so fee/byte
+      */
+    case object FeePerByte extends SortingOption
+
+    /**
+      * Sort transactions by fee paid for transaction contracts validation cost, so fee/execution unit
+      */
+    case object FeePerCostUnit extends SortingOption
+
+    /**
+      * @return randomly chosen mempool sorting strategy
+      */
+    def random(): SortingOption = {
+      if (Random.nextBoolean()) {
+        log.info("Sorting mempool by fee-per-byte")
+        FeePerByte
+      } else {
+        log.info("Sorting mempool by fee-per-cycle")
```

Comment:
Maybe better free this function from logging side-effect and move the message to the ouside also giving more context info to the message.
Especially when this method is used as default argument value of `empty` method. (see below)

Reply:
done